queue.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The go-zeromq Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package zmq4
+
+import (
+	"container/list"
+)
+
+const innerCap = 512
+
+type Queue struct {
+	rep *list.List
```

Comment:
AFAICT, we only use `Queue` as a queue of `Msg`.
couldn't we just copy-paste the code of `container/list` (or as an `internal` package) and adapt for this `Msg` data structure instead of paying the price of type-assertions?

(this could be done in a follow-up PR, but we could at least adapt `Queue`'s API to only accept `Msg` values)

Reply:
I anticipate this struct being used in the future for other types. But, indeed, it's currently only used to store `Msg` so I guess it should do just that. One can always make it generic whenever the need arises.

Though it uses `container/list`, it's not quite one. Each entry in the list is itself a fixed sized chunk that will contain multiple `Msg`. This is done for cache-friendliness, as a lot of the `Msg` will be located contiguously, and to amortize the cost of alloc (and pointers to be collected). The `innerCap` variable specifies the size of each chunk, which was determined empirically on an 8th Gen i7 CPU.